test/monarch/camp_monarch_interface.F90

```diff
@@ -218,741 +147,685 @@ function constructor(camp_config_file, interface_config_file, &
+            r=(k-1)*(I_E*I_N) + (j-1)*(I_E) + i-1
+            forall (i_spec = 1:size(this%map_monarch_id))
+              this%camp_state%state_var(this%init_conc_camp_id(i_spec)&
+              +r*state_size_per_cell) = this%init_conc(i_spec)
+            end forall
+            do i_spec=1, size(this%map_monarch_id)
+              MONARCH_conc(i,j,k,this%map_monarch_id(i_spec)) = &
+                this%camp_state%state_var(this%map_camp_id(i_spec))
+            end do
+            this%camp_state%state_var(this%gas_phase_water_id +(r*state_size_per_cell)) = &
+                    water_conc(i,j,k,WATER_VAPOR_ID) * &
+                            mwair / mwwat * 1.e6
+          end do
+        end do
+      end do
+    end if
   end subroutine get_init_conc
 
-!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
-
-  !> Get the MONARCH species names and indices (for testing only)
-  subroutine get_MONARCH_species(this, species_names, MONARCH_ids)
-
-    !> PartMC-camp <-> MONARCH interface
-    class(monarch_interface_t) :: this
-    !> Set of MONARCH species names
-    type(string_t), allocatable, intent(out) :: species_names(:)
-    !> MONARCH tracer ids
-    integer(kind=i_kind), allocatable, intent(out) :: MONARCH_ids(:)
 
-    species_names = this%monarch_species_names
-    MONARCH_ids = this%map_monarch_id
-
-  end subroutine get_MONARCH_species
-
-!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
-
-  !> Print the PartMC-camp data
-  subroutine do_print(this)
-
-    !> PartMC-camp <-> MONARCH interface
-    class(monarch_interface_t) :: this
-
-    call this%camp_core%print()
-
-  end subroutine do_print
-
-!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
-
-  !> Finalize the interface
   elemental subroutine finalize(this)
-
-    !> PartMC-camp <-> MONARCH interface
-    type(monarch_interface_t), intent(inout) :: this
-
-    if (associated(this%camp_core)) &
-            deallocate(this%camp_core)
-    if (associated(this%camp_state)) &
-            deallocate(this%camp_state)
-    if (allocated(this%monarch_species_names)) &
-            deallocate(this%monarch_species_names)
-    if (allocated(this%map_monarch_id)) &
-            deallocate(this%map_monarch_id)
-    if (allocated(this%map_camp_id)) &
-            deallocate(this%map_camp_id)
-    if (allocated(this%init_conc_camp_id)) &
-            deallocate(this%init_conc_camp_id)
-    if (allocated(this%init_conc)) &
-            deallocate(this%init_conc)
-    if (associated(this%species_map_data)) &
-            deallocate(this%species_map_data)
-    if (associated(this%init_conc_data)) &
-            deallocate(this%init_conc_data)
-    if (associated(this%property_set)) &
-            deallocate(this%property_set)
+    type(camp_monarch_interface_t), intent(inout) :: this
+    if (associated(this%camp_core)) deallocate(this%camp_core)
 
   end subroutine finalize
 
```

Comment:
why are these pointer deallocations removed?

Reply:
At some point it failed, but it seems to be fixed now, so I will left it as before.

Reply:
I found the error: Some variables like "init_conc_camp_id" are set only for rank 0, so it fails when using 40 MPI processes.